carmajava/cooperativemerge/src/main/java/gov/dot/fhwa/saxton/carma/plugins/cooperativemerge/ExecutionState.java

```diff
@@ -141,6 +148,16 @@ public void onMobilityOperationMessage(MobilityOperation msg) {
     double targetSpeed = Double.parseDouble(params.get(0));
     double maxAccel = Double.parseDouble(params.get(1));
     double targetSteer = Double.parseDouble(params.get(2));
+
+    //do a sanity check on the received values
+    if (targetSpeed < MIN_TARGET_SPEED  ||  targetSpeed > MAX_TARGET_SPEED  ||
+        maxAccel < MIN_ACCEL            ||  maxAccel > MAX_ACCEL  ||
+        targetSteer < MIN_TARGET_STEER  ||  targetSteer > MAX_TARGET_STEER) {
+      log.error("Received operation message with suspect strategy values. targetSpeed = " + targetSpeed +
+                ", maxAccel = " + maxAccel + ", targetSteer = " + targetSteer);
+      return;
```

Comment:
We need more than a return here. This class is the Execution state meaning the vehicle is actively being controlled by the RSU. I think there needs to be a notification of trajectory failure so cruising can take over and possibly a NACK message to the RSU.

Reply:
That was my feeling as well, but not understanding the overall logic flow well, I was just mimicking the behavior when the strategy string was totally hosed, such as at line 144.  The condition here is not really different than that one.  Same logic applies to your two other comments below on HoldingState and CommandingState.  I note that StandbyState returns a MobilityRequestResponse.NO_RESPONSE when it finds an error in the strategy string during its OnMobilityResponseMessage() method.  The OnMobilityOperationMessage() methods return void since RSUMeteringWorker's handleMobilityOperationMsg() just assumes the state transition will work.  I feel like adding that detection capability is an enhancement beyond the scope of this security issue.